fix: Rename optimistic update in children containers

ChrisChV opened the pull request:
## Description

- When you update the title of a unit/subsection in the subsection/section page, the text returns to the previous value for a while. This PR fix that issue
- Which edX user roles will this change impact? "Course Author.

## Supporting information

- This issue was encountered while working on https://github.com/openedx/frontend-app-authoring/issues/1971
- Internal ticket: [FAL-4165](https://tasks.opencraft.com/browse/FAL-4165)

## Testing instructions

- Go to the library home of a library
- Create one unit, one subsection and one section.
- Add the unit to the subsection.
- Add the subsection to the section.
- On the section page, rename the subsection, verify that the optimistic update works.
- On the subsection page, rename the unit, verify that the optimistic update works.


## Other information

- Fixing this issue results in an increase in requests, but the user  is not affected for the use of [placeHolderText](https://github.com/open-craft/frontend-app-authoring/pull/90/files#diff-1ae8b7fff8780b37c89ac8669f61267d687fa66f56a6640bf5ef3fe3e4c7f680R16)

ChrisChV committed:
fix: Rename optimistic update in children containers

ChrisChV committed:
style: Fix broken lint

pomegranited commented on src/library-authoring/containers/ContainerEditableTitle.tsx:
```diff
@@ -8,15 +8,29 @@ import messages from './messages';
 
-  const { data: container } = useContainer(containerId);
+  if (!readOnly) {
+    readOnly = libReadOnly;
```
Instead, should do this [below](https://github.com/open-craft/frontend-app-authoring/pull/90/commits/933f99d49a6ff0a78ab3c7afd355333cd27d68bf#diff-1ae8b7fff8780b37c89ac8669f61267d687fa66f56a6640bf5ef3fe3e4c7f680R62):

```
       readOnly={readOnly || libReadOnly}
```

ChrisChV replied:
Updated https://github.com/open-craft/frontend-app-authoring/pull/90/commits/42dfe1a4eff3360ff6e34ecdc905b59a52f38195

pomegranited reviewed (approved):
:+1: couple of lint issues but this is working well. I like the optimistic update behaviour, and would like to see it used everywhere.

I think we need a test or two though?

- [x] I tested this on my devstack with and without a running CMS, to make sure that the optimistic update reverts if it cannot be saved.
- [x] I read through the code
- [x] I checked for accessibility issues by using my keyboard to navigate
- [x] Includes documentation
- [x] User-facing strings are extracted for translation

ChrisChV committed:
test: Fix test for optimistic rename

ChrisChV committed:
fix: broken tests

ChrisChV committed:
fix: show unit published name in sidebar on content picker (#2100)

Fix the bug for show unit published name in sidebar on content picker.

ChrisChV committed:
fix: Refresh section list on subsection page (#2103)

Invalidates the query in the subsection page used to get the list of sections that contains the subsection

bradenmacdonald committed:
fix: searching "all courses" from studio home wasn't working (#2083)
